### PR TITLE
move to a separate file

### PR DESCRIPTION
**Pull request by wind57**

### CODE REVIEW

**Comment by wind57** [comment]
I'd like to become a contributor in this project also and as usual, my understanding of what it does and how starts from a few pull requests refactoring minor things. As such this PR does two things only:

- moves `SpringRetryConfig` to a separate class.
- removes `RetryContext` from `SpringRetryConfig`, as it is indeed not needed, how the comment said.

File: spring-cloud-circuitbreaker-spring-retry/src/main/java/org/springframework/cloud/circuitbreaker/springretry/SpringRetryConfigBuilder.java [hunk 1]
```diff
@@ -107,69 +106,4 @@ public SpringRetryConfig build() {
 		return config;
 	}
 
-	public static class SpringRetryConfig {
```
Comment (wind57) [inline spring-cloud-circuitbreaker-spring-retry/src/main/java/org/springframework/cloud/circuitbreaker/springretry/SpringRetryConfigBuilder.java:110]: this class has moved to a separate file.

File: spring-cloud-circuitbreaker-spring-retry/src/main/java/org/springframework/cloud/circuitbreaker/springretry/SpringRetryConfigBuilder.java [hunk 2]
```diff
@@ -107,69 +106,4 @@ public SpringRetryConfig build() {
...
-
-		private String id;
-
-		// TODO do we need this?
-		private RetryContext retryContext;
```
Comment (wind57) [inline spring-cloud-circuitbreaker-spring-retry/src/main/java/org/springframework/cloud/circuitbreaker/springretry/SpringRetryConfigBuilder.java:115]: this field and it's getters/setters where removed

**Comment by spencergibb** [comment]
Can you explain why this "moves SpringRetryConfig to a separate class?"

**Comment by wind57** [comment]
it's a public class residing in another public class, with package private methods. To me, it is a lot easier to read code. Making it an inner class does not provide any benefit at all, and in contrast defeats the purpose of an inner class.

imo, moving it to a separate file, makes the holder (`SpringRetryConfigBuilder`) a lot easier to reason about.

**Pull request merged**